apollo-backend/src/main/java/io/logz/apollo/controllers/StatusController.java

```diff
@@ -176,4 +177,9 @@ public String getLatestPodName(int environmentId, int serviceId, String groupNam
 
         return kubernetesHandler.getPodContainerNames(podName);
     }
+
+    @GET("/status/getUndeployedServicesInProductionEnvironments")
+    public List<Integer> getUndeployedServicesInProductionEnvironments() {
+       return serviceStatusHandler.getUndeployedServices();
```

Comment:
Thinking of your end game: should return environment+services (some kind of map).
Think about what you need - IDs, names, objects...

Reply:
Turn into a function and wrap with a dummy endpoint, add logic when you know what you need.